src/java/org/apache/cassandra/db/compaction/unified/Controller.java

```diff
@@ -596,10 +598,15 @@ public static Map<String, String> validateOptions(Map<String, String> options) t
             try
             {
                 long sizeInBytes = FBUtilities.parseHumanReadableBytes(s);
+                // zero is a valid option to disable featue
                 if (sizeInBytes < 0)
-                    throw new ConfigurationException(String.format("Invalid configuration, %s should be positive: %s",
-                                                                   MIN_SSTABLE_SIZE_OPTION,
-                                                                   s));
+                    throw new ConfigurationException(String.format("Invalid configuration, %s should be greater than or equal to 0 (zero)",
+                                                                   MIN_SSTABLE_SIZE_OPTION));
+                int limit = (int) Math.ceil(targetSSTableSize * INVERSE_SQRT_2);
+                if (sizeInBytes >= limit )
+                    throw new ConfigurationException(String.format("Invalid configuration, %s (%sB) should be less than: %sB",
```

Comment:
Nit: Let's add "the target size minimum" after "less than" to make sure the user knows where that number comes from.

Also, make both values human readable, by applying `FBUtilities.prettyPrintMemory()` (see e.g. the exception on line 500).

Reply:
will be taken care of on commit

Reply:
What kind of invalid value? Could you give an example?

Reply:
I understand the problem now, but it's not something I personally would mind with the current phrasing of the exception.

When told e.g. "less than 707.106MiB", wouldn't people try "707.105MiB"?